test/formik.test.tsx

```diff
@@ -963,5 +983,37 @@ describe('<Formik>', () => {
 
       expect((tree.instance() as any).resetForm).toHaveBeenCalledWith('data');
     });
+
+    it('should reset dirty flag even if initialValues has changed', async () => {
+      const tree = mount(<WithState />);
+      expect(tree.find(Form).props().dirty).toEqual(false);
+
+      tree
+        .find(Form)
+        .find('input')
+        .simulate('change', {
+          persist: noop,
+          target: {
+            id: 'name',
+            value: 'Ian',
+          },
+        });
+
+      expect(tree.find(Form).props().dirty).toEqual(true);
+
+      tree.setState({ data: { name: 'Jared' } });
+
+      await tree
+        .find(Form)
+        .props()
+        .handleReset();
+
+      expect(
+        tree
+          .update()
```

Comment:
IMO should work without update call here, but it doesn't. Cant get why.

Reply:
because https://github.com/airbnb/enzyme/blob/master/docs/guides/migration-from-2-to-3.md#for-mount-updates-are-sometimes-required-when-they-werent-before